### ISSUE TITLE
Nested Count statement with predicate is translated to COUNT(*)

### ISSUE THREAD

**Issue by prodenx** [aspnet/EntityFrameworkCore#18882]
I have query with several Counts nested in Select statement after GroupBy. Generated SQL does not have conditions in COUNT statement and returns wrong result

### Steps to reproduce

``` C#
 _repository.Where(product => product.Date >= date).GroupBy(product => product.ShiftId).OrderBy(shiftGroup => shiftGroup.Key).Select(shiftGroup =>
                    new
                    {
                        A = shiftGroup .Count(i => i.ReviewStateId == ProductReviewState.Good),
                        B = shiftGroup .Count(i =>
                            i.ReviewStateId != ProductReviewState.NotReviewed &&
                            i.ReviewStateId != ProductReviewState.Skipped),
                        C = shiftGroup .Count()
                    }).ToListAsync()
```
Fields A, B and C have same value in result.

Generated SQL

``` SQL
SELECT COUNT(*) AS [A], COUNT(*) AS [B], COUNT(*) AS [C]
FROM [Products] AS [s]
WHERE (([s].[Date] >= @__Date_0)
GROUP BY [s].[ShiftId]
ORDER BY [s].[ShiftId]
```

### Further technical details

EF Core version: v3.1.0-preview2
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: NET Core 3.1
Operating system: Windows 10
IDE: Visual Studio 2019 16.3.8

**Comment by hamedmaleki1366** [comment]
also for me : [](https://github.com/aspnet/EntityFrameworkCore/issues/18836#issue-520789207)

**Comment by smitpatel** [comment]
Fixed in #18553
